mlos_bench/mlos_bench/environments/script_env.py

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Base scriptable benchmark environment.
+"""
+
+import abc
+import re
+from typing import Dict, Iterable, Optional
+
+from mlos_bench.environments.base_environment import Environment
+from mlos_bench.services.base_service import Service
+from mlos_bench.tunables.tunable_groups import TunableGroups
+
+
+class ScriptEnv(Environment, metaclass=abc.ABCMeta):
+    """
+    Base Environment that runs scripts for setup/run/teardown.
+    """
+
+    _RE_INVALID = re.compile(r"[^a-zA-Z0-9_]")
+
+    def __init__(self,
+                 *,
+                 name: str,
+                 config: dict,
+                 global_config: Optional[dict] = None,
+                 tunables: Optional[TunableGroups] = None,
+                 service: Optional[Service] = None):
+        """
+        Create a new environment for script execution.
+
+        Parameters
+        ----------
+        name: str
+            Human-readable name of the environment.
+        config : dict
+            Free-format dictionary that contains the benchmark environment
+            configuration. Each config must have at least the `tunable_params`
+            and the `const_args` sections. It must also have at least one of
+            the following parameters: {`setup`, `run`, `teardown`}.
+            Additional parameters:
+                * `script_params` - an array of parameters to pass to the script
+                  as shell environment variables, and
+                * `script_params_rename` - a dictionary of {to: from} mappings
+                  of the script parameters. If not specified, replace all
+                  non-alphanumeric characters with underscores.
+            If neither `script_params` nor `script_params_rename` are specified,
+            pass *all* parameters to the script.
+        global_config : dict
+            Free-format dictionary of global parameters (e.g., security credentials)
+            to be mixed in into the "const_args" section of the local config.
+        tunables : TunableGroups
+            A collection of tunable parameters for *all* environments.
+        service: Service
+            An optional service object (e.g., providing methods to
+            deploy or reboot a VM, etc.).
+        """
+        super().__init__(name=name, config=config, global_config=global_config,
+                         tunables=tunables, service=service)
+
+        self._script_setup = self.config.get("setup")
+        self._script_run = self.config.get("run")
+        self._script_teardown = self.config.get("teardown")
+
+        self._script_params: Optional[Iterable[str]] = self.config.get("script_params")
+        self._script_params_rename: Dict[str, str] = self.config.get("script_params_rename", {})
```

Comment:
These also don't look like they match the schema names.
The end to end test script I mentioned would probably help catch this.